server/src/main/java/org/opensearch/common/lucene/uid/VersionsAndSeqNoResolver.java

```diff
@@ -165,6 +170,18 @@ public static DocIdAndVersion loadDocIdAndVersion(IndexReader reader, Term term,
             PerThreadIDVersionAndSeqNoLookup lookup = lookups[leaf.ord];
             DocIdAndVersion result = lookup.lookupVersion(term.bytes(), loadSeqNo, leaf);
             if (result != null) {
+                if (result.version > 0 && currentCriteria != null) {
+                    SegmentReader unwrappedReader = (SegmentReader) (FilterLeafReader.unwrap(leaf.reader()));
+                    String prevCriteria = unwrappedReader.getSegmentInfo().info.getAttribute(CriteriaBasedCodec.BUCKET_NAME);
+                    assert prevCriteria != null;
+                    if (prevCriteria.equals(currentCriteria) == false) {
+                        throw new ImmutableCriteriaException(
+                            "Updating grouping criteria is not allowed for context aware enabled indices",
+                            null
```

Comment:
This only throws error if the bucket is changed (e.g. 4xx can't be changed to 5xx but a 404 can be changed to 429 if buckets are defined as 4xx & 5xx) We should document this behaviour.